connector_salesforce/unit/rest_api_adapter.py

```diff
@@ -0,0 +1,347 @@
+        """Try to get a connexion using password policy
+        :return: A simplesalesforce connexion
+        :rtype: :py:class:`simplesalesforce.Salesforce
+        """
+        with error_handler(self.backend_record):
+            sf = Salesforce(
+                instance_url=self.backend_record.url,
+                username=self.backend_record.username,
+                password=self.backend_record.password,
+                security_token=self.backend_record.security_token,
+            )
+        return sf
+
+    def _sf_from_oauth2(self):
+        """Try to get a connexion using OAuth2 policy
+        :return: A simplesalesforce connexion
+        :rtype: :py:class:`simplesalesforce.Salesforce
+        """
+        with error_handler(self.backend_record):
+            sf = Salesforce(
+                instance_url=self.backend_record.url,
+                session_id=self.backend_record.consumer_token,
+                sandbox=self.backend_record.sandbox
+            )
+        return sf
+
+    def _sf_from_organization_id(self):
+        """Try to get a connexion using oragnisation uuid policy
+        :return: A simplesalesforce connexion
+        :rtype: :py:class:`simplesalesforce.Salesforce
+        """
+        with error_handler(self.backend_record):
+            sf = Salesforce(
+                username=self.backend_record.username,
+                password=self.backend_record.password,
+                organizationId=self.backend_record.organization_uuid
+            )
+        return sf
+
+    def get_sf_connection(self):
+        """Try to get a connexion based on current backend
+        :return: A simplesalesforce connexion
+        :rtype: :py:class:`simplesalesforce.Salesforce
+        """
+        assert self.backend_record, 'Backend record not available'
+        if self.backend_record.authentication_method == 'oauth2':
+            return self._sf_from_oauth2()
+        elif self.backen_record.authentication_method == 'pwd_token':
+            return self._sf_from_login_password()
+        elif self.backend_record.authentication_method == 'ip_filtering':
+            return self._sf_from_organization_id()
+        else:
+            raise NotImplementedError('Authentication method not supported')
+
+    def get_updated(self, start_datetime_str=None, end_datetime_str=None):
+        """Get a list of updated record on Salesforce using REST API
+        If not date passed if will do a SOQL query to
+        retrieve all existing and active record.
+        :param start_datetime_str: an Odoo datetime string to lookup
+                                   on start time
+
+        :param stop_datetime_str: an Odoo datetime string to lookup
+                                   on stop time
+
+        :return: generator of updated ids
+        :rtype: generator
+        """
+        if start_datetime_str:
+            if not end_datetime_str:
+                end_datetime_str = '2100-01-01 00:00:00'
```

Comment:
Hello, yes I used it in the first time but Salesforce API was not happy with it and gave me an invalide date error